### PR TITLE
[DSK-26] add authentication & authorization middlewares

### PR DESCRIPTION
**Pull request by jan-zabloudil**

### CODE REVIEW

File: transport/ping.go
Comment (jan-zabloudil) [inline transport/ping.go:1]: Added to transport/general.go

File: transport/response.go
Comment (jan-zabloudil) [inline transport/response.go:1]: Slightly refactored - more responses added, logging centralized in `WriteErrorResponse`

File: utils/background.go
Comment (jan-zabloudil) [inline utils/background.go:1]: Not used in this PR. 
But part of shared functionality used in following branches (for example for sending email in background).

File: utils/validate.go [hunk 1]
```diff
@@ -19,3 +21,13 @@ func init() {
 	translator, _ = uni.GetTranslator("en")
 	_ = entrans.RegisterDefaultTranslations(Validate, translator)
 }
+
+func TranslateValidationErrs(errs validator.ValidationErrors) map[string]string {
```
Comment (jan-zabloudil) [inline utils/validate.go:25]: Again, used in following branches.

**Comment by CermakM** [comment]
Added some commens to the #4 that are still relevant in this PR I believe 👍

**Pull request merged**